### DISCUSSION TITLE
Cannot read properties of undefined (reading 'timeoutType')

### DISCUSSION THREAD

**Discussion by emeralddd** [nodejs/undici#1633] [Q&A]
```
/home/obziappdev/wheat/node_modules/undici/lib/client.js:1167
        if (socket[kParser].timeoutType !== TIMEOUT_IDLE) {
                            ^

TypeError: Cannot read properties of undefined (reading 'timeoutType')
    at _resume (/home/obziappdev/wheat/node_modules/undici/lib/client.js:1167:29)
    at resume (/home/obziappdev/wheat/node_modules/undici/lib/client.js:1131:3)
    at connect (/home/obziappdev/wheat/node_modules/undici/lib/client.js:1116:3)
    ```
When i Started my Discordjs bot it caused this

nodejs: 16.17.0

**Comment by mcollina** [comment]
Thanks for reporting!

Can you provide steps to reproduce? We often need a [reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), e.g. some code that allows someone else to recreate your problem by just copying and pasting it. If it involves more than a couple of different file, create a new repository on GitHub and add a link to that.

**Reply by CherryDT** [reply]
Interestingly, this can even happen [with the version bundled to nodejs itself](https://stackoverflow.com/questions/73745614/fetch-json-api-in-node-js-v18).

**Comment by SalamPS** [comment]
> ```
> /home/obziappdev/wheat/node_modules/undici/lib/client.js:1167
>         if (socket[kParser].timeoutType !== TIMEOUT_IDLE) {
>                             ^
> 
> TypeError: Cannot read properties of undefined (reading 'timeoutType')
>     at _resume (/home/obziappdev/wheat/node_modules/undici/lib/client.js:1167:29)
>     at resume (/home/obziappdev/wheat/node_modules/undici/lib/client.js:1131:3)
>     at connect (/home/obziappdev/wheat/node_modules/undici/lib/client.js:1116:3)
>     ```
> When i Started my Discordjs bot it caused this
> 
> nodejs: 16.17.0
> ```

Did it fixed yet? I'm facing the same problem r.n

**Comment by wonderkidshihab** [comment]
Any solution yet? Still having this issue...

```
node_modules/undici/lib/client.js:1191
        if (socket[kParser].timeoutType !== TIMEOUT_IDLE) {
                            ^

TypeError: Cannot read properties of undefined (reading 'timeoutType')
    at _resume (/home/artifiti/discord-bot/node_modules/undici/lib/client.js:1191:29)
    at resume (/home/artifiti/discord-bot/node_modules/undici/lib/client.js:1155:3)
    at connect (/home/artifiti/discord-bot/node_modules/undici/lib/client.js:1140:3)

Node.js v19.4.0
```

**Reply by lmclarke** [reply]
No solution I'm aware of. Same issue, driving me insane.

**Reply by ronag** [reply]
Latest master might have a solution. I'll try to make a new release later today.

**Reply by ronag** [reply]
I've just published undici@5.15.2. Try with that.

**Reply by lmclarke** [reply]
> I've just published undici@5.15.2. Try with that.

The error changed to this with undici@5.15.2:

```
TypeError: Cannot destructure property 'exports' of 'undefined' as it is undefined.
    at new Parser (/home/tfzmud/tfz-bot/node_modules/undici/lib/client.js:420:34)
    at connect (/home/tfzmud/tfz-bot/node_modules/undici/lib/client.js:1082:23)
```

Another file

```
(node:9094) ExperimentalWarning: stream/web is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
/home/tfzmud/tfz-bot/node_modules/undici/lib/client.js:420
  constructor (client, socket, { exports }) {
                                 ^

TypeError: Cannot destructure property 'exports' of 'undefined' as it is undefined.
    at new Parser (/home/tfzmud/tfz-bot/node_modules/undici/lib/client.js:420:34)
    at connect (/home/tfzmud/tfz-bot/node_modules/undici/lib/client.js:1082:23)
```

**Reply by ronag** [reply]
Can you try again with v5.16.0?

**Reply by lmclarke** [reply]
> Can you try again with v5.16.0?

> node deploy-commands.js

```
Started refreshing 3 application (/) commands.
(node:25348) ExperimentalWarning: stream/web is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
/home/tfzmud/tfz-bot/node_modules/undici/lib/client.js:355
  return await WebAssembly.instantiate(mod, {
                           ^

RangeError: WebAssembly.instantiate(): Out of memory: wasm memory
    at lazyllhttp (/home/tfzmud/tfz-bot/node_modules/undici/lib/client.js:355:28)
```

**Reply by ronag** [reply]
There you have it. Out of memory. You don't have enough memory to initiate the wasm code.

**Reply by ronag** [reply]
What node version are you using?

**Reply by ronag** [reply]
PR is welcome if you can figure out how to reduce the memory requirements. Also you want want to look into what other modules you are loading that might take up memory.

**Reply by lmclarke** [reply]
> What node version are you using?

v16.9.0. Also attempted with v16.18.1.

**Reply by lmclarke** [reply]
> PR is welcome if you can figure out how to reduce the memory requirements. Also you want want to look into what other modules you are loading that might take up memory.

No idea really. Nothing but discord.js and its dependencies.

**Reply by ronag** [reply]
Unfortunately, I can't help more than this atm. At least we get the correct error now.

**Reply by lmclarke** [reply]
No worries, thank you for the attempt.
